quinn/src/endpoint.rs

```diff
@@ -214,11 +217,12 @@ impl Endpoint {
         let addr = socket.local_addr()?;
         let socket = self.runtime.wrap_udp_socket(socket)?;
         let mut inner = self.inner.state.lock().unwrap();
+        inner.abandoned_socket = Some(inner.socket.clone());
         inner.socket = socket;
```

Comment:
We can avoid a `clone` here and clarify intent while we're at it:
```suggestion
        inner.abandoned_socket = Some(mem::replace(&mut inner.socket, socket));
```